File: my-folder/0219-contains-duplicate-ii/solution.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer,Integer> integerMap=new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            if(integerMap.containsKey(nums[i]))
            {
                int value=integerMap.get(nums[i]);
                if(Math.abs(value-i)<=k)
                    return true;
                else
                    integerMap.put(nums[i],i);
                
            }
            else 
               integerMap.put(nums[i],i); 
        }
        return false;
    }
}

File: my-folder/0219-contains-duplicate-ii/solution.kt
class Solution {
    fun containsNearbyDuplicate(nums: IntArray, k: Int): Boolean {
        val numSet=mutableSetOf<Int>()
        for(i in nums.indices)
        {
            if(i>k)
              numSet.remove(nums[i-k-1])
            if(!numSet.add(nums[i])) return true
        }
        return false
    }
}
